File: src/Services/Catalog/Catalog.Api/Application/Catalog/GetCatalog/GetCatalogResult.cs
using BuildingBlocks.Pagination;
using Catalog.Api.Domain.Entities;

namespace Catalog.Api.Application.Catalog.GetCatalog;

public record GetCatalogResult(int PageIndex, int PageSize, long Count, List<CatalogEntity> Products)
{
    public int PageIndex { get; set; } = PageIndex;
    public int PageSize { get; set; } = PageSize;
    public long Count { get; set; } = Count;
    public List<CatalogEntity> Products { get; set; } = Products;
}

File: uis/WebApp/src/app/shared/models/get-catalog-result.ts
import {Catalog} from '@app/shared/models/catalog';

export class GetCatalogResult {
  pageIndex: number;
  pageSize: number;
  count: number;
  products: Catalog[];

  constructor(pageIndex: number, pageSize: number, count: number, products: Catalog[]) {
    this.pageIndex = pageIndex;
    this.pageSize = pageSize;
    this.count = count;
    this.products = products;
  }
}
